### PR TITLE
Update pom's to meet the requirements for uploading to maven central

### PR DESCRIPTION
**Pull request by SamBarker**
### Type of change

- Enhancement / new feature

### Description

To be able to release the public API through maven central the POM's are must pass central validation.

### Additional Context

To reduce the [lottery factor](https://blog.devgenius.io/building-a-good-lottery-factor-bf67f7de04d8) for the project I've also included some documentation on how to actually publish the release.

### CODE REVIEW

File: docs/Releasing.adoc [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+. `mvn deploy -Prelease -DskipTests=true -DreleaseSigningKey=<YOUR_KEY_SHORT_ID>`
+.. The `release` profile enables GPG signing
+.. The release profile will not run without specifying the `releaseSigningKey` property.
+. Validate the release in the central Nexus instance.
+.. Locate the https://central.sonatype.org/publish/release/#locate-and-examine-your-staging-repository[Staging repository]
+.. Check that the right tree of artefacts are there, ensure there are source & docs jars.
+// .. *TODO*: Using the staging repository to validate the release.
+. `Close` the staging repository in the central Nexus instance, see the https://central.sonatype.org/publish/release/#close-and-drop-or-release-your-staging-repository[Central release docs]. This triggers the promotion from staging to being publicly available on central.
+. Switch to the next version number for development.
+.. `export DEV_VERSION=0.1.0`
```
Comment (robobario) [inline docs/Releasing.adoc:49]: should this be a `-SNAPSHOT` version?

File: docs/Releasing.adoc [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+== Running a release
+
+.Releasing Locally
+. `mvn clean install`
+.. We are looking for a clean and green build.
+.. If it's not, shout scream and swear.
+.* Think *very very* carefully if its worth proceeding. Releases are forever!
+. Switch to the release version number.
+.. `export RELEASE_VERSION=0.1.0`
+.. `mvn versions:set -DnewVersion=$\{RELEASE_VERSION}`
```
Comment (robobario) [inline docs/Releasing.adoc:33]: After this step the property is:
```
        <kroxyliciousApiVersion>0.1.0-SNAPSHOT</kroxyliciousApiVersion>
```
but the api artifacts are at `0.1.0`

File: pom.xml [hunk 3]
```diff
@@ -526,5 +604,49 @@
...
+                                <id>enforce-property</id>
+                                <goals>
+                                    <goal>enforce</goal>
+                                </goals>
+                                <configuration>
+                                    <rules>
+                                        <requireProperty>
+                                            <property>releaseSigningKey</property>
+                                            <message>You must specify your GPG key when running a release</message>
+                                        </requireProperty>
```
Comment (robobario) [inline pom.xml:642]: suggest adding:

```
<requireReleaseDeps>
    <message>No Snapshots Allowed!</message>
</requireReleaseDeps>
```

Comment (robobario) [reply pom.xml:642]: alternatively:

[banDynamicVersions](https://maven.apache.org/enforcer/enforcer-rules/banDynamicVersions.html) - bans all dependencies requiring version resolution at build time (i.e. version ranges, placeholders RELEASE/LATEST or SNAPSHOT versions).

Comment (robobario) [reply pom.xml:642]: maybe this too: [requireReleaseVersion](https://maven.apache.org/enforcer/enforcer-rules/requireReleaseVersion.html) - enforces that the artifact is not a snapshot.

File: docs/Releasing.adoc (hunk 2 above)
Comment (SamBarker) [reply docs/Releasing.adoc:33]: Hmm good spot.

Comment (SamBarker) [reply docs/Releasing.adoc]: fixed in 9d02b10a8854c6732538e85b0748064f8d2a9f5f

**Review by k-wall — APPROVED** [review]

**Pull request merged**